# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../directfb/dfb_types.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur DirectFB-0.9.23.orig/include/dfb_types.h DirectFB-0.9.23/include/dfb_types.h
--- DirectFB-0.9.23.orig/include/dfb_types.h	2005-07-26 23:58:50.000000000 +0300
+++ DirectFB-0.9.23/include/dfb_types.h	2005-10-20 14:19:01.000000000 +0300
@@ -8,24 +8,40 @@
 #define __u8  u_int8_t
 #define __u16 u_int16_t
 #define __u32 u_int32_t
-#define __u64 u_int64_t
+#ifdef __x86_64__
+	#define __u64 unsigned long long int
+#else
+	#define __u64 u_int64_t
+#endif
 
 #define __s8  int8_t
 #define __s16 int16_t
 #define __s32 int32_t
-#define __s64 int64_t
+#ifdef __x86_64__
+	#define __s64 long long int
+#else
+	#define __s64 int64_t
+#endif
 
 #else
 
 #define __u8  uint8
 #define __u16 uint16
 #define __u32 uint32
-#define __u64 uint64
+#ifdef __x86_64__
+	#define __u64 unsigned long long int
+#else
+	#define __u64 uint64
+#endif
 
 #define __s8  sint8
 #define __s16 sint16
 #define __s32 sint32
-#define __s64 sint64
+#ifdef __x86_64__
+	#define __s64 long long int
+#else
+	#define __s64 sint64
+#endif
 
 #endif
 
